Input output traffic stats and command process count for each client.

minchopaskal opened the pull request:
We already have global stats for input traffic, output traffic and how many commands have been executed.

However, some users have the difficulty of locating the IP(s) which have heavy network traffic. So here some stats for single client are introduced.
```
tot-net-in   // Total network input bytes read from the client
tot-net-out  // Total network output bytes sent to the client
tot-cmds     // Total commands the client has executed
```
These three stats are shown in `CLIENT LIST` and `CLIENT INFO`.

Discussion about this was started as redis PR: https://github.com/redis/redis/pull/12640 and was merged in valkey with: https://github.com/valkey-io/valkey/pull/327

Co-authored-by Chen Tianjie <TJ_Chen@outlook.com>

minchopaskal commented:
@sundb please review

minchopaskal commented:
@sundb maybe it's not a bad idea to run benchmarks, since as valkey PR states the changes are in the hotpath. If so I guess we will do it in an official PR

sundb commented on src/blocked.c:
```diff
@@ -6,6 +6,8 @@
  * Licensed under your choice of the Redis Source Available License 2.0
  * (RSALv2) or the Server Side Public License v1 (SSPLv1).
  *
+ * Portions of this file are available under BSD3 terms; see REDISCONTRIBUTIONS for more information.
```
missing Valkey block
```
 * Copyright (c) 2009-Present, Redis Ltd.
 * All rights reserved.
 *
 * Copyright (c) 2024-present, Valkey contributors.
 * All rights reserved.
 *
 * Licensed under your choice of the Redis Source Available License 2.0
 * (RSALv2) or the Server Side Public License v1 (SSPLv1).
 *
```

sundb commented on src/networking.c:
```diff
@@ -3157,9 +3162,13 @@ sds catClientInfoString(sds s, client *client) {
+        " tot-cmds=%U", client->commands_processed));
 
     if (paused) resumeIOThread(client->running_tid);
+
```
```suggestion
```

sundb commented on src/server.c:
```diff
@@ -3818,8 +3818,14 @@ void call(client *c, int flags) {
         }
     }
 
-    if (!(c->flags & CLIENT_BLOCKED))
```
~~miss the license for this file.~~
blind.

minchopaskal replied:
yep it was already there

minchopaskal committed:
Input output traffic stats and command process count for each client.

We already have global stats for input traffic, output traffic and how
many commands have been executed.

However, some users have the difficulty of locating the IP(s) which have
heavy network traffic. So here some stats for single client are
introduced.
```
tot-net-in   // Total network input bytes read from the client
tot-net-out  // Total network output bytes sent to the client
tot-cmds     // Total commands the client has executed
```
These three stats are shown in `CLIENT LIST` and `CLIENT INFO`.

Though the metrics are handled in hot paths of the code, personally I
don't think it will slow down the server. Considering all other complex
operations handled nearby, this is only a small and simple operation.

However we do need to be cautious when adding more and more metrics, as
discussed in redis/redis#12640, we may need to find a way to tell
whether this has obvious performance degradation.

Co-authored-by Chen Tianjie <TJ_Chen@outlook.com>

minchopaskal committed:
Fix race condition in test for per-client in/out/cmds stats

minchopaskal committed:
Uncomment wait_for_blocked_client

minchopaskal committed:
Fix output bytes for client info
